remote-api/src/main/java/org/alfresco/rest/api/impl/DownloadsImpl.java

```diff
@@ -172,4 +198,99 @@ private Download getStatus(NodeRef downloadNodeRef)
+     * A limit can be applied to prevent large sized requests preventing the asynchronous call to start.
+     * The cache is used to prevent duplication of checks, as it is possible to provide a folder and its contents as 
+     * separate nodes in the download request.
+     * 
+     * @param nodeRefs
+     * @param checkLimit The maximum number of nodes to check, set to -1 for no limit
+     * @param cache Tracks nodes that we have already checked, if null an empty cache will be created
+     */
+    @Experimental
+    private void checkArchiveStatus(NodeRef[] nodeRefs, int checkLimit, List<NodeRef> cache)
+    {
+        // Create the cache for recursive calls.
+        if (cache == null) 
+        {
+            cache = new ArrayList<NodeRef>();
+        }
+
+        var folders = new ArrayList<NodeRef>();
+        for (NodeRef nodeRef : nodeRefs) 
+        {
+            // We hit the number of nodes we want to check.
+            if (cache.size() == checkLimit) 
+            {
+                logger.info(
+                        String.format(
+                                "Maximum check of %d reached for archived content. No more checks will be performed and download will still be created.",
+                                checkLimit));
+                return;
+            }
+            // Already checked this node, we can skip.
+            if (cache.contains(nodeRef)) 
+            {
+                continue;
+            }
+
+            QName qName = nodeService.getType(nodeRef);
+            if (qName.equals(ContentModel.TYPE_FOLDER))
+            {
+                // We'll check the child nodes at the end in case there are other nodes in this loop that is archived.
+                folders.add(nodeRef); 
+            }
+            else if (qName.equals(ContentModel.TYPE_CONTENT))
+            {
+                Map<String, String> props = contentService.getStorageProperties(nodeRef, qName);
+                if (!props.isEmpty() && Boolean.valueOf(props.get(ObjectStorageProps.X_ALF_ARCHIVED.getValue())))
+                {
+                    throw new ArchivedIOException("One or more nodes' content is archived and not accessible.");
+                }
+            }
+            cache.add(nodeRef); // No need to check this node again.
+        }
+
+        // We re-run the folder contents at the end in case we hit content that is archived in the first loop and can stop early.
+        for (NodeRef nodeRef : folders) 
+        {
+            NodeRef[] childRefs = nodeService.getChildAssocs(nodeRef).stream()
+                                                                     .map(childAssoc -> childAssoc.getChildRef())
+                                                                     .toArray(NodeRef[]::new);
+            checkArchiveStatus(childRefs, checkLimit, cache); // We'll keep going until we have no more folders in children.
+        }
+    }
+
+    @Experimental
+    protected boolean canCheckArchived()

```

Comment:
Do we need this extra method (could we move up into protected "checkArchiveStatus" ?

Reply:
Doesn't look like it, I'll move in the next commit. I think I was keep my options open for unit tests but never used it.

Reply:
I have since refactor and re-added this method. The new implementation recommended by @mpichura is better sat in it's own method.